tests/cnf/ran/powermanagement/internal/helper/helper.go

```diff
@@ -77,6 +81,98 @@ func SetPowerModeAndWaitForMcpUpdate(perfProfile *nto.Builder, node nodes.Builde
 	return err
 }
 
+// SetCPUFreq updates the performance profile with the given isolated and reserved
+// core frequencies and verifies that the frequencies have been updated on the spoke cluster.
+func SetCPUFreq(
+	perfProfile *nto.Builder,
+	desiredIsolatedCoreFreq *performancev2.CPUfrequency,
+	desiredReservedCoreFreq *performancev2.CPUfrequency) error {
+	glog.V(tsparams.LogLevel).Infof("Set Reserved and Isolated CPU Frequency on performance profile")
+
+	// Update PerfProfile with new CPU Frequencies
+	perfProfile.Definition.Spec.HardwareTuning.IsolatedCpuFreq = desiredIsolatedCoreFreq
+	perfProfile.Definition.Spec.HardwareTuning.ReservedCpuFreq = desiredReservedCoreFreq
+	_, err := perfProfile.Update(true)
+
+	if err != nil {
+		return err
+	}
+
+	// Determine isolated CPU number from IsolatedCPUset
+	isolatedCPUSet, err := cpuset.Parse(string(*perfProfile.Object.Spec.CPU.Isolated))
+	if err != nil {
+		return err
+	}
+
+	isolatedCPUsList := isolatedCPUSet.List()
+	isolatedCPUNumber := isolatedCPUsList[0]
+
+	spokeCommand := fmt.Sprintf("cat /sys/devices/system/cpu/cpufreq/policy%v/scaling_max_freq |cat -",
+		isolatedCPUNumber)
+
+	// Wait for Isolated CPU Frequency to be updated.
+	err = wait.PollUntilContextTimeout(
+		context.TODO(), 5*time.Second, 1*time.Minute, true, func(ctx context.Context) (bool, error) {
+			if err != nil {
+				return false, err
+			}
+			// Get current isolated core frequency from spoke cluster and compare to desired frequency
+			consoleOut, err := cluster.ExecCommandOnSNO(raninittools.Spoke1APIClient, 3, spokeCommand)
+
+			if err != nil {
+				return false, errors.New(fmt.Sprintf("Command failed: %v", consoleOut))
+			}
+
+			currIsolatedCoreFreq, err := strconv.Atoi(strings.TrimSpace(consoleOut))
+
+			if err != nil {
+				return false, errors.New(fmt.Sprintf("string conversion failed failed: %v", err))
+			}
+
+			if currIsolatedCoreFreq != int(*desiredIsolatedCoreFreq) {
+				return false, errors.New("current Isolated CPU Frequency does not match expected frequency")
```

Comment:
returning false with a non-nil error fails the `wait.PollUntilContextTimeout` so this might need to be `return false, nil`

Reply:
@josclark42 In this case it's not about `fmt.Errorf` vs `errors.New`, it's that returning an error from `wait.PollUntilContextTimeout` will prevent it from trying again after the interval. I think in this case you still want to keep checking if the current is equal to the desired so you should return a nil error. Otherwise the function will only run once.

Reply:
Got it now. Thanks for the explanation! Fixed.